build.sbt

```diff
@@ -1344,25 +1341,25 @@ lazy val `scio-smb`: Project = project
       "com.google.protobuf" % "protobuf-java" % protobufVersion,
       "com.spotify" %% "magnolify-parquet" % magnolifyVersion,
       "joda-time" % "joda-time" % jodaTimeVersion,
-      "org.apache.avro" % "avro" % avroVersion,
       "org.apache.beam" % "beam-sdks-java-core" % beamVersion,
       "org.apache.beam" % "beam-sdks-java-extensions-protobuf" % beamVersion,
       // #3260 work around for sorter memory limit until we patch upstream
       // "org.apache.beam" % "beam-sdks-java-extensions-sorter" % beamVersion,
       "org.apache.beam" % "beam-sdks-java-io-google-cloud-platform" % beamVersion excludeAll (testLibs: _*),
       "org.apache.beam" % "beam-sdks-java-io-hadoop-common" % beamVersion,
       "org.apache.beam" % "beam-vendor-guava-26_0-jre" % beamVendorVersion,
-      "org.apache.parquet" % "parquet-avro" % parquetVersion excludeAll ("org.apache.avro" % "avro"),
-      "org.apache.parquet" % "parquet-common" % parquetVersion,
-      "org.apache.hadoop" % "hadoop-common" % hadoopVersion,
-      "org.apache.parquet" % "parquet-column" % parquetVersion,
-      "org.apache.parquet" % "parquet-hadoop" % parquetVersion,
       "org.checkerframework" % "checker-qual" % checkerFrameworkVersion,
-      "org.tensorflow" % "tensorflow-core-api" % tensorFlowVersion,
       "org.scala-lang.modules" %% "scala-collection-compat" % scalaCollectionCompatVersion,
       "org.slf4j" % "slf4j-api" % slf4jVersion,
       // provided
       "com.github.ben-manes.caffeine" % "caffeine" % caffeineVersion % Provided,
+      "org.apache.avro" % "avro" % avroVersion % Provided,
+      "org.apache.hadoop" % "hadoop-common" % hadoopVersion % Provided,
+      "org.apache.parquet" % "parquet-avro" % parquetVersion % Provided excludeAll ("org.apache.avro" % "avro"),
+      "org.apache.parquet" % "parquet-column" % parquetVersion % Provided,
+      "org.apache.parquet" % "parquet-common" % parquetVersion % Provided,
+      "org.apache.parquet" % "parquet-hadoop" % parquetVersion % Provided,
+      "org.tensorflow" % "tensorflow-core-api" % tensorFlowVersion % Provided,
```

Comment:
For smb, I think it is safe to assume pple will also have `scio-avro` `scio-parquet` or `scio-tensorflow` as depenedecy. Do not pull everything here

Reply:
hm I do think it's fairly common to have scio-smb but not scio-parquet in your dependencies; I think this change makes sense, but let's definitely call it out in the migration guide.

Reply:
In theory we can wrap `ClassNotFoundException` with more meaningful instructions. Even with good docs, there will be confused users writing to support channels

Reply:
I added some checks in the static class/object initialization of the various SMB impl